### PR TITLE
Add supportfor `IntanRecordingExtractor` "one-file-per-signal" and "one-file-per-channel" formats

### PR DESCRIPTION
**Pull request by h-mayorquin**
As stated on the title. This is possible because we merged the following neo:

https://github.com/NeuralEnsemble/python-neo/pull/1402

As this translated cleanly the only thing that I am adding here are the tests.

### CODE REVIEW

**Comment by h-mayorquin** [comment]
Interesting this fails here. It works locally for me but it is hard to know which of the tests is actually failing because of the structure of the tests with entities.

**Comment by h-mayorquin** [comment]
Ok, the error was solved. The reason was that the tests were not using the latests versions of neo and probeinterface. That is, it seems that these lines are not working for the full tests:

https://github.com/SpikeInterface/spikeinterface/blob/2983463b655a6f10130b8017913f2ca90d4fd9fe/pyproject.toml#L153-L156

Probably something cache related. It is good for you to be aware of this @samuelgarcia @alejoe91

File: pyproject.toml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "spikeinterface"
-version = "0.101.0"
+version = "0.101.0.dev0"
```
Comment (alejoe91) [inline pyproject.toml:3]: Can you revert this? See other PR 
```suggestion
version = "0.101.0"
```

**Review by alejoe91 — CHANGES_REQUESTED** [review]
Thanks @h-mayorquin 

Looks good, except for the change in version

**Review by alejoe91 — APPROVED** [review]

**Comment by alejoe91** [comment]
> Thanks @h-mayorquin
> 
> Looks good, except for the change in version

Changed

**Pull request merged**